uvicorn/supervisors/multiprocess.py

```diff
@@ -72,3 +73,18 @@ def shutdown(self) -> None:
             click.style(str(self.pid), fg="cyan", bold=True)
         )
         logger.info(message, extra={"color_message": color_message})
+
+    def monitor(self) -> None:
+        while not self.should_exit.is_set():
+            time.sleep(0.5)
+            if self.should_exit.is_set():
+                break
+            # Restart expired workers.
+            for process in self.processes:
+                if not process.is_alive():
```

Comment:
This is not reliable. We can't know if the process is stuck. It will only give the false impression of a good process manager.

Reply:
I've considered this but I don't think it's reasonable to expect Uvicorn to have this level of involvement because if a worker is hung/stuck, the resolution could be quite different for every project: It's up to the developer to decide when/how to kill a "stuck" worker.

The ideal situation is for Uvicorn to mimic k8s here- and only become involved when the worker is dead/crashed/exited, which is what `is_alive()` does perfectly.

Reply:
This is a good change. I agree with @gnat. I think the feature that @Kludex is after is a slightly different scope.

Reply:
On the description it says: "we're eliminating an entire dependency with only a few lines.", but gunicorn does check if the process is stuck, so it's the same scope, and this is not a replacement for it.